# Apriori

In [1]:
# Movie Recommendation: Suport(M) = #user watchlist containing(M)/ #user watchlist
#                       Confidence(M1-->M2) = #user watchlist containing(M1 and M2)/ #user watchlist containing(M1)
#                       Lift(M1-->M2) = Confidence/Support

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from apyori import apriori

In [3]:
df = pd.read_csv('/Users/omkarmutreja/Downloads/Apriori_Python/Market_Basket_Optimisation.csv',header=None)
print(df.shape)
df.head()

(7501, 20)


0          1           2                 3             4   \
0         shrimp    almonds     avocado    vegetables mix  green grapes   
1        burgers  meatballs        eggs               NaN           NaN   
2        chutney        NaN         NaN               NaN           NaN   
3         turkey    avocado         NaN               NaN           NaN   
4  mineral water       milk  energy bar  whole wheat rice     green tea   

                 5     6               7             8             9   \
0  whole weat flour  yams  cottage cheese  energy drink  tomato juice   
1               NaN   NaN             NaN           NaN           NaN   
2               NaN   NaN             NaN           NaN           NaN   
3               NaN   NaN             NaN           NaN           NaN   
4               NaN   NaN             NaN           NaN           NaN   

               10         11     12     13             14      15  \
0  low fat yogurt  green tea  honey  salad  mineral water  salmon   
1             NaN        NaN    NaN    NaN            NaN     NaN   
2             NaN        NaN    NaN    NaN            NaN     NaN   
3             NaN        NaN    NaN    NaN            NaN     NaN   
4             NaN        NaN    NaN    NaN            NaN     NaN   

                  16               17       18         19  
0  antioxydant juice  frozen smoothie  spinach  olive oil  
1                NaN              NaN      NaN        NaN  
2                NaN              NaN      NaN        NaN  
3                NaN              NaN      NaN        NaN  
4                NaN              NaN      NaN        NaN

In [4]:
transactions =[]
for i in range(7501):
    transactions.append([str(df.values[i,j]) for j in range(20)])

In [5]:
rules = apriori(transactions,min_support=0.003,min_confidence=0.2,min_lift=3,min_length=2)
results = list(rules)
results[:1] # Top 1 rule

[RelationRecord(items=frozenset({'chicken', 'light cream'}), support=0.004532728969470737, ordered_statistics=[OrderedStatistic(items_base=frozenset({'light cream'}), items_add=frozenset({'chicken'}), confidence=0.29059829059829057, lift=4.84395061728395)])]

## We see that people who buy light cream have a 29% chance of buying chicken as well.